### PR TITLE
remove not used globalCacheReader

### PR DESCRIPTION
**Pull request by nioshield**
Signed-off-by: nio <nioshield@gmail.com>

### What problem does this PR solve?
Close https://github.com/chaos-mesh/chaos-mesh/issues/3332

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [ ] release-2.2
  - [ ] release-2.1

### Checklist

CHANGELOG

- [ ] I have updated the `CHANGELOG.md`
- [ ] I have labeled this PR with "no-need-update-changelog"

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ For more information and how-to, see [RFC: Keep A Changelog](https://github.com/
 - Removed unused local function from statuscheck/manager.go [#3228](https://github.com/chaos-mesh/chaos-mesh/pull/3228)
 - Removed ui build and test for arm64 [#3305](https://github.com/chaos-mesh/chaos-mesh/pull/3305)
 - Removed not used clientset in cmd/chaos-controller-manager/main.go [#3334](https://github.com/chaos-mesh/chaos-mesh/pull/3334)
+- Removed not used globalCacheReaderRemoved in cmd/chaos-controller-manager/provider/controller.go [#3343](https://github.com/chaos-mesh/chaos-mesh/pull/3343)
```
Comment (z1cheng) [inline CHANGELOG.md:38]: A typo? Maybe should delete the suffix "Removed" in "globalCacheReaderRemoved",  like "Removed not used globalCacheReader"

**Comment by STRRL** [comment]
Welcome @nioshield , thanks for your contribution!

**Comment by STRRL** [comment]
> How do I fix IOChaos errors in merge check? @STRRL worriedworried

don't worry, the e2e tests are flaky, I would retry it;

**Review by YangKeao — APPROVED** [review]
LGTM

**Comment by STRRL** [comment]
> /cc @YangKeao

It seems taichi bot does not work as expected, a little weird. 🤔🤔

**Pull request merged**